crates/rolldown/src/stages/generate_stage/code_splitting.rs

```diff
@@ -295,4 +278,76 @@ impl<'a> GenerateStage<'a> {
       });
     });
   }
+
+  fn apply_advanced_chunks(
+    &mut self,
+    module_to_bits: &IndexVec<ModuleIdx, BitSet>,
+    module_to_assigned: &mut IndexVec<ModuleIdx, bool>,
+    chunk_graph: &mut ChunkGraph,
+  ) {
+    // `ModuleGroup` is a temporary representation of `Chunk`. A valid `ModuleGroup` would be converted to a `Chunk` in the end.
+    struct ModuleGroup {
+      modules: Vec<ModuleIdx>,
+    }
+    let Some(chunking_options) = &self.options.advanced_chunks else {
+      return;
+    };
+
+    let Some(mut match_groups) =
+      chunking_options.groups.as_ref().map(|inner| inner.iter().collect::<Vec<_>>())
+    else {
+      return;
+    };
+
+    if match_groups.is_empty() {
+      return;
+    }
+
+    // Higher priority group goes first.
+    match_groups.sort_by_key(|group| Reverse(group.priority));
```

Comment:
Not sure if two groups `[a, b]` has same priority, look like we should take `a` first?

Reply:
Notice this is a stable sort. If the priority is the same, then we just pick the first one.